cowamm/kpi/competitors/uni_swap_style/largest_uni_style_kpis_4304295.sql

```diff
@@ -0,0 +1,101 @@
+-- Computes volume, tvl and APR for Uni Swap style pools (Uni, Pancake, Sushi)
+-- APR is measured as the fees earned per $ invested, over the last 24 hours, projected over 1 year
+-- The type of pool (Uni/Pancake/Sushi) is not specified
+-- Parameters:
+-- {{blockchain}}: The blockchain to query
+-- {{number_of_pools}}: The number of largest pools to return
+-- {{end_time}}: The end time of the time window (end_time - 1 day; end_time), defaults to now()
+
+-- select the pool with the largest latest k
+with pool as (
+    select
+        pool_address as contract_address,
+        project,
+        token0,
+        token1,
+        tvl
+    from "query_4303563(blockchain='{{blockchain}}', number_of_pools = '{{number_of_pools}}')"
+),
+
+syncs as (
+    select
+        pool.*,
+        tx_hash as evt_tx_hash,
+        index as evt_index,
+        block_time as evt_block_time,
+        block_number as evt_block_number,
+        varbinary_to_uint256(substr(data, 1, 32)) as reserve0,
+        varbinary_to_uint256(substr(data, 33, 32)) as reserve1,
+        rank() over (partition by (logs.contract_address) order by block_time desc) as latest
+    from {{blockchain}}.logs
+    inner join pool
+        on logs.contract_address = pool.contract_address
+    where
+        block_time >= date_add('day', -1, (case when '{{end_time}}' = '2024-01-01' then now() else timestamp '{{end_time}}' end))
```

Comment:
Here we should just use a default value that is very far in the future and not a date that we may actually want to query